ath79: small fixes wr1043nd-v4

lucize opened the pull request:
fix sysupgrade check
move usb to v4 dts because v5 doesn't have it
make wan mac address behave like ar71xx target
switch tplink-safeloader to dynamic partitions

add orange wan led support, it can be userspace activated like

on:
`echo default-on > /sys/class/leds/tp-link\:orange\:wan/trigger`
off:
`echo none > /sys/class/leds/tp-link\:orange\:wan/trigger`

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

lucize committed:
ath79: small fixes wr1043nd-v4

fix sysupgrade check
move usb to v4 dts because v5 doesn't have it
make wan mac address behave like ar71xx target

add orange wan led support, it can be userspace activated like

on:
echo default-on > /sys/class/leds/tp-link\:orange\:wan/trigger

off:
echo none > /sys/class/leds/tp-link\:orange\:wan/trigger

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

juppin commented on target/linux/ath79/dts/qca9563_tl-wr1043nd-v4.dts:
```diff
@@ -10,3 +10,16 @@
+&usb0 {
+	status = "okay";
+
+	hub_port0: port@1 {
```
What about the trigger-sources of usb led? This will throw an error if anyone will use qca9563_tl-wr1043n.dtsi and not have hub_port0.

lucize replied:
you are right

lucize commented:
on ar71xx the wan port has +1 to mac address, but on the device there is only one mac address, should I increment it ?

juppin commented:
> on ar71xx the wan port has +1 to mac address, but on the device there is only one mac address, should I increment it ?

I don´t see a +1.
Which wan port? V4 and V5 does use the same mac for eth0 (wan and lan) and wmac...

lucize commented:
```
        tl-wr1043nd-v4)
                lan_mac=$(mtd_get_mac_binary product-info 8)
                wan_mac=$(macaddr_add "$lan_mac" 1)
                ;;

```

lucize commented:
updated the wan mac address to behave like ar71xx target

lucize commented:
also updated support for wan orange led so it can be triggered in userspace when wan is down

juppin commented:
@lucize Probably you could also switch to dynamic partition layout for kernel and rootfs as with this commit https://github.com/openwrt/openwrt/commit/97da92b1277a40fa7daf354ed52b8ea25e06a779

lucize commented:
@juppin I'll test later

lucize commented:
@juppin @mkresin 
I tried like this

```
diff --git a/tools/firmware-utils/src/tplink-safeloader.c b/tools/firmware-utils/src/tplink-safeloader.c
index 1c34406251..fc62736b88 100644
--- a/tools/firmware-utils/src/tplink-safeloader.c
+++ b/tools/firmware-utils/src/tplink-safeloader.c
@@ -860,15 +860,10 @@ static struct device_info boards[] = {
                .support_trail = '\x00',
                .soft_ver = NULL,

-               /**
-                   We use a bigger os-image partition than the stock images (and thus
-                   smaller file-system), as our kernel doesn't fit in the stock firmware's
-                   1MB os-image.
-               */
+               /** We're using a dynamic kernel/rootfs split here */
                .partitions = {
                        {"fs-uboot", 0x00000, 0x20000},
-                       {"os-image", 0x20000, 0x200000},
-                       {"file-system", 0x220000, 0xd30000},
+                       {"firmware", 0x20000, 0xf30000},
                        {"default-mac", 0xf50000, 0x00200},
                        {"pin", 0xf50200, 0x00200},
                        {"product-info", 0xf50400, 0x0fc00},

```

```
[    0.320646] m25p80 spi0.0: w25q128 (16384 Kbytes)
[    0.325571] 7 fixed-partitions partitions found on MTD device spi0.0
[    0.332127] Creating 7 MTD partitions on "spi0.0":
[    0.337101] 0x000000000000-0x000000020000 : "u-boot"
[    0.342891] 0x000000020000-0x000000f50000 : "firmware"
[    0.388944] 2 tplink-fw partitions found on MTD device firmware
[    0.395094] 0x000000020000-0x0000001a3b51 : "kernel"
[    0.400843] 0x0000007a0000-0x000000f50000 : "rootfs"
[    0.406609] mtd: device 3 (rootfs) set to be root filesystem
[    0.412481] mtdsplit: no squashfs found in "rootfs"
[    0.417586] 0x000000f50000-0x000000f70000 : "product-info"
[    0.423862] 0x000000f70000-0x000000fc0000 : "config"
[    0.429674] 0x000000fc0000-0x000000fd0000 : "partition-table"
[    0.436288] 0x000000fd0000-0x000000ff0000 : "logs"
[    0.441827] 0x000000ff0000-0x000001000000 : "ART"
[    1.896476] jffs2_scan_eraseblock(): End of filesystem marker found at 0x10000
[    1.903954] jffs2_build_filesystem(): unlocking the mtd device...
[    1.904041] done.
[    1.912430] jffs2_build_filesystem(): erasing all blocks after the end marker...
[    2.534275] random: fast init done
[...]
[   30.893135] done.
[   30.904720] jffs2: notice: (1) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   30.922381] VFS: Mounted root (jffs2 filesystem) readonly on device 31:3.
[...]
[   30.947293] Kernel panic - not syncing: No working init found.  Try passing init= option to kernel. See Linux Documentation/admin-guide/init.rst for guidance.
[   30.962474] Rebooting in 1 seconds..
```

lucize commented:
That fixed it

lucize committed:
firmware-utils: tplink-safeloader: switch tl-wr1043nd-v4 to dynamic partitions

Use the new dynamic partition split in tplink-safeloader so we no longer
have to worry about kernel size increases.

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

mkresin commented:
@lucize you need to change the image build code for the ar71xx tl-wr1043nd-v4 image as well!

Any chance to test the dynamic partition thingy with an upgrade from the stock tp-link firmware (factory image)?

lucize commented:
I don't have a stock image for flashing and recovery seems not to work, I'll make one, I think I have to cut the bootloader from it, no ?
I'll change the ar71xx

adrianschmutzler commented:
With 1043v4 and v5, you can just do TFTP with the image from TP-Link support pages.

lucize commented:
@adrianschmutzler I tried holding the reset button but no tft recovery initiated and writing from uboot did not work for me (maybe the wiki is not accurate for v4/5), I can only recover from initramfs image, is safe to use sysupgrade from there ?

adrianschmutzler commented:
@lucize I cannot answer your question, but I normally just do:
https://wiki.openwrt.org/toh/tp-link/tl-wr1043nd#flashingrecovery_using_tftp_only_without_serial_console

Image name for V5 is: wr1043v5_tp_recovery.bin (So without "n" or "nd")
Maybe try all possible versions for the v4:
wr1043v4_tp_recovery.bin
wr1043nv4_tp_recovery.bin
wr1043ndv4_tp_recovery.bin

You can just use the Image from TP-Link, e.g.
https://www.tp-link.com/de/download/TL-WR1043N.html#Firmware

I'm not a pro in those things, so I can help you beyond that. But simple TFTP for the "ar71xx" devices never failed me ...

lucize commented:
it seems you don't have to press reset button before powering up the device, but just after, so I could restore to original firmware
@mkresin the factory imaged worked as it should

mkresin commented:
@lucize If you add and test the small change I send you via mail, I'm fine to merge the PR.

lucize commented:
right away

lucize committed:
ath79: tl-wr1043nd-v4 usb led reference

use label for leds so it can be referenced by the node in dts

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

mkresin commented:
Thanks! Merged to [my staging tree](https://git.openwrt.org/?p=openwrt/staging/mkresin.git;a=summary).
